### PR TITLE
MN-002/Implement-Join-Room

### PR DESCRIPTION
**Pull request by williamking1221**
Added Join Room -> Form to Input Room ID and Name -> Validation of Room ID -> Player Waiting Page (Boilerplate)

Moved stuff around w.r.t. file structure and URLs

Note: URL style may be susceptible to a cunning player to go to /rooms/:roomId/game-master and see that dashboard...need to add something like playerID to the URL with will be a randomly generated ID and unique to a game.

Added TODO comments for future

### CODE REVIEW

File: src/firebase/database.js [hunk 1]
```diff
@@ -28,3 +28,15 @@ export const deleteGame = (roomKey) => {
...
+
+export const doesGameExist = async (roomKey) => {
+  try {
+    const roomRef = ref(database, `activeGames/${roomKey}`);
+    const snapshot = await get(roomRef);
+    console.log(snapshot.exists());
+    return snapshot.exists();
+  } catch (error) {
+    console.error('Error checking room:', error);
+    throw error;
```
Comment (Octophi) [inline src/firebase/database.js:40]: Do you know what the throw here actually does?

Comment (williamking1221) [reply src/firebase/database.js:40]: Passes the error to the file that calls it so it logs?

Comment (Octophi) [reply src/firebase/database.js:40]: Hmm so you're logging at both levels? I guess that's fine

Comment (williamking1221) [reply src/firebase/database.js:40]: I can take it out as well

Comment (Octophi) [reply src/firebase/database.js:40]: Also I was being dumb, forgot this is just JavaScript and not firebase specific lol

**Review by Octophi — APPROVED** [review]

**Pull request merged**